db/version_set.cc

```diff
@@ -3471,6 +3487,25 @@ Status VersionSet::Recover(
         break;
       }
     }
+    // use persistent_stats_version_edit to restore column family for the
+    // first time
+    if (mutable_db_options_->stats_persist_period_sec != 0 &&
+        !persistent_stats_cfd_exists) {
+      ColumnFamilyOptions persistent_stats_options;
+      VersionEdit persistent_stats_cf_edit;
+      persistent_stats_cf_edit.AddColumnFamily(kPersistentStatsColumnFamilyName);
+      uint32_t next_id = column_family_set_->GetNextColumnFamilyID();
+      persistent_stats_cf_edit.SetColumnFamily(next_id);
+      ColumnFamilyData* persistent_stats_cfd =
+          CreateColumnFamily(persistent_stats_options, &persistent_stats_cf_edit);
+      // In recovery, nobody else can access it, so it's fine to set it to be
+      // initialized earlier.
+      persistent_stats_cfd->set_initialized();
+      BaseReferencedVersionBuilder* builder =
+          new BaseReferencedVersionBuilder(persistent_stats_cfd);
+      builder->version_builder()->Apply(&persistent_stats_cf_edit);
+      builders.insert({next_id, builder});
```

Comment:
Can we reuse existing CreateColumnFamily() code? Actually do we need to do it in VersionSet? Can we do this in DBImpl after VersionSet::Recover() returns?

Reply:
Reusing `DBImpl::CreateColumnFamilyImpl` makes sense because we support dynamic changing of `stats_persist_period_sec` and we also support dynamic adding/dropping column families.
Therefore, `DBImpl::Recover` can check whether `stats_persist_period_sec` is non-zero AND the `stats` CF has been created during `VersionSet::Recover`. If not, create the column family. We also need an API. When the user changes the `stats_persist_period_sec` from 0 to non-zero, we call `DBImpl::CreateColumnFamilyImpl`.